extensions/algebra/circuit/src/extension/modular.rs

```diff
@@ -713,19 +713,18 @@ pub fn find_non_qr(modulus: &BigUint, rng: &mut impl Rng) -> BigUint {
         // since 2^((p-1)/2) = (-1)^((p^2-1)/8)
         BigUint::from_u8(2u8).unwrap()
     } else {
-        let mut non_qr = rng.gen_biguint_range(
-            &BigUint::from_u8(2).unwrap(),
-            &(modulus - BigUint::from_u8(1).unwrap()),
-        );
+        let mut buf = vec![0u8; modulus.to_bytes_be().len()];
+        let lower = BigUint::from_u8(2).unwrap();
+        let upper = modulus - BigUint::one();
+        rng.fill_bytes(&mut buf);
+        let mut non_qr = BigUint::from_bytes_be(&buf) % &upper + &lower;
```

Comment:
noting this is not a uniform sampling; the previous [implementation](https://docs.rs/num-bigint/latest/src/num_bigint/bigrand.rs.html#113) seems to use rejection sampling. It is not strictly required here, but I would suggest switching to it.